src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  checkHealth() {
+    if (this.health === 0) {
+      Animal.alive = Animal.alive.filter(animal => animal.health > 0);
+    }
+  }
```

Comment:
it is better to make this method to be a static method of Animal class since we're filtering static property.

make it like this:

```javascript
Animal.checkHealth = function() { //...
``` 

also, we omit additional check in this case
```suggestion
  checkHealth() {
      Animal.alive = Animal.alive.filter(animal => animal.health >= 0);
  }
```